### ISSUE TITLE
[dashboard] Unify workspace (instance) visuals across dashboard

### ISSUE THREAD

**Issue by geropl** [gitpod-io/gitpod#8155]
At the moment we copy'n'paste code for displaying a workspace (instance) and their state in various places, leading[ to incosistencies](https://github.com/gitpod-io/gitpod/pull/8125#pullrequestreview-880089702).

We already have code in place for the [status indicator](https://github.com/gitpod-io/gitpod/blob/5ff578ed67e45f62932c6ac148324f015b9104ae/components/dashboard/src/workspaces/WorkspaceEntry.tsx#L201), for instance, but don't use it everywhere.

Ideally, the different parts (status indicator, unpushed changes, etc.) would be implemented in recombinable functions, so we can still vary the resulting look.

regarding unpushed changes: If there are no changes, we can probably skip that element instead of displaying "No changes".

**Comment by geropl** [comment]
@gtsiolis You were _seconds_ faster! :laughing: 

But we should really have a bot for this... :confused:

**Comment by axonasif** [comment]
> @geropl: But we should really have a bot for this... 😕

Do you mean a bot for adding labels and such 👀 ?

**Comment by gtsiolis** [comment]
> But we should really have a bot for this... 😕

@geropl I agree, [`gitbot`](https://github.com/gitpod-io/gitbot) has great potential in helping in such triage operations. Usually, inferring team labels from component labels sounds optimal.
